packages/docgen/lib/get-jsdoc-from-token.js

```diff
@@ -40,6 +41,20 @@ module.exports = ( token ) => {
 							: tag.description,
 				};
 			} );
+
+			if ( jsdoc.tags.length === 0 ) {
+				const potentialTypeAnnotation = getTypeAnnotation(
+					{ tag: 'type' },
+					token,
+					0
+				);
+				if ( potentialTypeAnnotation !== '' ) {
+					jsdoc.tags.push( {
+						tag: 'type',
```

Comment:
Why doesn't jsdoc.tags include tags for `@type` already? (if I'm reading this right, it looks like it _does_ include tags for the other things we support?)

Reply:
So, it does, but `@type` is only used to annotate JSDoc types. When you're in TypeScript world, you'll never use the `@type` annotation because you'll always just do `foo: Type`.

So we push a dummy `@type` for the Intermediate Representation part of `docgen` to use to generate the IR which in turn can generate the markdown.

Basically we need to fake the `@type` annotation for docgen's sake by extracting the type from TypeScript when we can.

If `potentialTypeAnnotation` is an empty string, that means we couldn't extract the type for some reason. This could be as simple as an undocumented exported constant like a string or something. In those cases we just leave it alone and allow the regular documentation to take place without adding the type annotation to it.

Reply:
Ahhh that totally makes sense. Thanks for explaining that!